src/components/OSCALCatalogGroups.js

```diff
@@ -52,32 +53,69 @@ const ComponentTab = styled(Tab)(({ theme }) => ({
   textTransform: "none",
 }));
 
-function a11yProps(index, title) {
+function a11yProps(groupId) {
   return {
-    id: `vertical-tab-${index}`,
-    "aria-controls": `vertical-tabpanel-${title}`,
+    id: `vertical-tab-${groupId}`,
+    "aria-controls": `vertical-tab-${groupId}`,
   };
 }
 
 TabPanel.propTypes = {
-  index: PropTypes.number.isRequired,
-  value: PropTypes.number.isRequired,
+  value: PropTypes.string.isRequired,
 };
 
 export default function OSCALCatalogGroups(props) {
-  const [value, setValue] = React.useState(0);
+  const { groups, urlFragment } = props;
+  const [openTab, setOpenTab] = React.useState(groups[0]?.id);
+  const [controlListItemOpened, setControlListItemOpened] =
+    React.useState(false);
 
   const handleChange = (event, newValue) => {
-    setValue(newValue);
+    setOpenTab(newValue);
   };
 
+  const handleFragment = useCallback(() => {
+    // Initially set item opened to false for new fragment to be handled
+    setControlListItemOpened(false);
+    // Grab fragment identifier following hash character if fragment exists in location
+    const controlFragment = urlFragment !== "" ? urlFragment : null;
+    // Find catalog group fragment
+    const controlGroupingFragment =
+      determineControlGroupFromFragment(controlFragment);
+    // Determine higher-level control or sub-control exists within catalog group
+    const catalogControl =
+      groups
+        ?.find((group) => group.id === controlGroupingFragment)
+        ?.controls?.find((control) => control.id === controlFragment) ||
+      groups
+        ?.find((group) => group.id === controlGroupingFragment)
+        ?.controls?.find((control) => controlFragment?.includes(control.id))
+        ?.controls?.find((subcontrol) => subcontrol.id === controlFragment);
+    if (!catalogControl) {
+      return;
+    }
+    // Confirm catalog tab group can be grabbed
+    const elementWithFragment = document.getElementById(
+      `vertical-tab-${controlGroupingFragment}`
+    );
+    if (elementWithFragment) {
+      setOpenTab(controlGroupingFragment);
+    }
+  }, [urlFragment, groups]);
```

Comment:
How is this different from the code in `OSCALCatalogGroup.js`? It seems like there's a lot of repetition. Can the code here be reusable? Or at least broken into chunks or composed?

Reply:
A lot of the section has been changed in order to support fragment changes to how the group is captured. Some of this may look similar such as grabbing an element and opening a tab however these involve different states and require a different method of finding the controls. I've done a little bit of refactoring here and made a reusable function for fragment validation.